web_app/tests/test_positions.py

```diff
@@ -0,0 +1,282 @@
+        "supply_price": 100,
+        "debt_price": 200,
+    }
+    with (
+        patch(
+            "web_app.contract_tools.mixins.deposit.DepositMixin.get_repay_data"
+        ) as mock_get_repay_data,
+        patch(
+            "web_app.db.crud.PositionDBConnector.get_contract_address_by_wallet_id"
+        ) as mock_get_contract_address,
+        patch(
+            "web_app.db.crud.PositionDBConnector.get_position_id_by_wallet_id"
+        ) as mock_get_position_id,
+    ):
+        mock_get_repay_data.return_value = mock_repay_data
+        mock_get_contract_address.return_value = "mock_contract_address"
+        mock_get_position_id.return_value = 123
+        mock_get_repay_data.return_value = mock_repay_data
+        DepositMixin.get_repay_data = mock_get_repay_data
+        response = client.get(
+            f"/api/get-repay-data?supply_token={supply_token}&wallet_id={wallet_id}"
+        )
+        mock_get_contract_address.assert_called_once_with(wallet_id)
+        mock_get_position_id.assert_called_once_with(wallet_id)
+        mock_get_repay_data.assert_called_once_with(supply_token)
+        expected_response = {
+            **mock_repay_data,
+            "contract_address": "mock_contract_address",
+            "position_id": "123",
+        }
+        assert response.status_code == 200
+        assert response.json() == expected_response
+
+
+@pytest.mark.anyio
+async def test_get_repay_data_missing_wallet_id(client: AsyncClient) -> None:
+    """
+    Test for missing wallet ID when attempting to retrieve repayment data, which should return a 404 error.
+
+    Args:
+        client (AsyncClient): The test client for the FastAPI application.
+
+    Returns:
+        None
+    """
+    supply_token = "valid_supply_token"
+    wallet_id = ""
+    with (
+        patch(
+            "web_app.contract_tools.mixins.deposit.DepositMixin.get_repay_data"
+        ) as mock_get_repay_data,
+        patch(
+            "web_app.db.crud.PositionDBConnector.get_contract_address_by_wallet_id"
+        ) as mock_get_contract_address,
+        patch(
+            "web_app.db.crud.PositionDBConnector.get_position_id_by_wallet_id"
+        ) as mock_get_position_id,
+    ):
+        mock_get_repay_data.return_value = None
+        mock_get_contract_address.side_effect = None
+        mock_get_position_id.side_effect = None
+        response = client.get(
+            f"/api/get-repay-data?supply_token={supply_token}&wallet_id={wallet_id}"
+        )
+        assert response.status_code == 404
+        assert response.json() == {"detail": "Wallet not found"}
+
+
+# Test cases for /api/create_position_with_transaction_data
+@pytest.mark.anyio
+async def test_create_position_success(client: AsyncClient) -> None:
+    """
+    Test for successfully creating a position with valid form data.
+
+    Args:
+        client (AsyncClient): The test client for the FastAPI application.
+
+    Returns:
+        None
+    """
+    form_data = {
```

Comment:
Please, use: `pytest.mark.parametrize` and add more than only one example of data. This data is very important, we want to make sure, that we covered as much as possible cases